okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -493,6 +493,14 @@ private void onGoAway(int lastKnownStreamId, Status status) {
     boolean notifyShutdown;
     synchronized (lock) {
       notifyShutdown = !goAway;
+    }
+    if (notifyShutdown) {
+      // We call transportShutdown() here, because once the goAway becomes true, other thread may
+      // call stopIfNecessary() and cause listener.transportTerminated() be called.
```

Comment:
Actually this code is wrong, if two threads call into onGoAway(), then transportShutdown() may be called twice, will figure out another way.